sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1434,6 +1434,10 @@ class Analyzer(
                 // We get an aggregate function, we need to wrap it in an AggregateExpression.
                 case agg: AggregateFunction => AggregateExpression(agg, Complete, isDistinct)
                 // This function is not an aggregate function, just return the resolved one.
+                case multiply: Multiply if multiply.left.dataType == CalendarIntervalType =>
```

Comment:
This check is still not solving the issue. Where should I put this conversion logic?

Reply:
Please move this into `TypeCoercion.DateTimeOperations`.

Reply:
Also add a test to TypeCoercionSuite. You can use the `rule for date/timestamp operations` test.